scripts/release/commit_goa/reactome2go.pl

```diff
@@ -1,100 +1,110 @@
+$pass ||= $GKB::Config::GK_DB_PASS;
+$port ||= $GKB::Config::GK_DB_PORT;
+
+my $dba= GKB::DBAdaptor->new(
+    -user   => $user,
+    -host   => $host,
+    -pass   => $pass,
+    -port   => $port,
+    -dbname => $db,
+    -DEBUG => $debug
+);
+
+if (!$output_file) {
+    $output_file = 'Reactome2GoV' . get_version($db);
+}
 
-$opt_db || die "Need database name (-db).\n";
-#$opt_date || die "Need date (-date).\n";  #need to revisit this, at present some instances don't have InstanceEdits attached, this should be fixed
+open my $file, '>', $output_file;
+binmode $file, ':encoding(UTF-8)';
+binmode STDOUT, ':encoding(UTF-8)';
+
+foreach my $event (@{$dba->fetch_instance(-CLASS => 'Event')}) {
+    print "$PROGRAM_NAME: event->db_id=" . $event->db_id() . "\n";
+    foreach my $line (get_reactome_2_go_mapping_lines($event)) {
+        print {$file} "$line\n";
+    }
+}
+close $file;
 
-my $dba= GKB::DBAdaptor->new
-    (
-     -user   => $opt_user || '',
-     -host   => $opt_host,
-     -pass   => $opt_pass,
-     -port   => $opt_port,
-     -dbname => $opt_db,
-     -DEBUG => $opt_debug
-     );
+print "$PROGRAM_NAME has finished its job\n";
 
-$opt_db =~ /(\d+)$/;
-my $outfile = "Reactome2GoV" . $1;
+sub get_version {
+    my $db_name = shift;
 
-open(FILE, ">$outfile") or die "$0: could not open file $outfile";
-binmode(FILE, ":utf8");
-my $ar = $dba->fetch_instance(-CLASS => 'Event'); # Obtains a reference to the array of all Reactome events
+    my ($version) = $db_name =~ /(\d+)$/msx;
+    $version ||= $db_name;
 
-# Each event in Reactome is processed
-binmode(STDOUT, ":utf8");
-foreach my $ev (@{$ar}) {
-    print "$0: ev->db_id=" . $ev->db_id() . "\n";
-    Reactome2GO($ev);
+    return $version;
 }
 
-close(FILE); # The output file has all entries now and is closed
-
-print "$0 has finished its job\n";
-
-
-
-sub Reactome2GO {
-	my $event = shift;
-	my @go;
-	push @go, $event->GoBiologicalProcess->[0];
-	
-	foreach my $cat (@{$event->CatalystActivity}) {
-		push @go, $cat->Activity->[0];	
-	}
-	
-	return unless $event->stableIdentifier->[0];
-	my $eventid = $event->stableIdentifier->[0]->identifier->[0];
-	
-	my $eventdisplayname = $event->_displayName->[0];
-	$eventdisplayname =~ s/\s+/ /g;
-	$eventdisplayname = trim($eventdisplayname);
-	
-	my $eventspecies = $event->species->[0]->name->[0];
-	
-	foreach my $go (@go) {
-		next unless $go;
-		my $goid;
-		my $goname;
-		if ($go->accession) {
-			$goid = $go->accession->[0];
-		}
-		if ($go->_displayName) {
-			$goname = $go->_displayName->[0];
-		}
-	
-		print FILE "Reactome:$eventid $eventdisplayname, $eventspecies > GO:$goname ; GO:$goid \n";
-	}
+sub get_reactome_2_go_mapping_lines {
+    my $event = shift;
+
+    if (!$event->stableIdentifier->[0]) {
+        return;
+    }
+
+    my $event_id = $event->stableIdentifier->[0]->identifier->[0];
+    (my $event_displayname = trim($event->displayName)) =~ s/\s+/ /msxg; # Removes extra whitespace from displayName
```

Comment:
Yes, they are -- the parentheses ensure the assignment happens first and then the search and replace.  Otherwise, it would try to do the search and replace on the result of the trim function rather than the value stored `$event_displayname`.  It's yet another Perlism :).